app/models/dataset/DatasetService.scala

```diff
@@ -164,16 +168,17 @@ class DatasetService @Inject()(organizationDAO: OrganizationDAO,
   private def updateDataSource(
       dataStore: DataStore,
       dataSource: InboxDataSource,
-      foundDatasets: Map[String, List[Dataset]]
+      foundDatasetsByDirectoryName: Map[String, List[Dataset]]
   )(implicit ctx: DBAccessContext): Fox[Option[ObjectId]] = {
-    val foundDatasetOpt = foundDatasets.get(dataSource.id.name).flatMap(_.headOption)
+    val foundDatasetOpt = foundDatasetsByDirectoryName.get(dataSource.id.directoryName).flatMap(_.headOption)
     foundDatasetOpt match {
       case Some(foundDataset) if foundDataset._dataStore == dataStore.name =>
         updateKnownDataSource(foundDataset, dataSource, dataStore).toFox.map(Some(_))
       case Some(foundDataset) => // This only returns None for Datasets that are present on a normal Datastore but also got reported from a scratch Datastore
         updateDataSourceDifferentDataStore(foundDataset, dataSource, dataStore)
       case _ =>
-        insertNewDataset(dataSource, dataStore).toFox.map(Some(_))
+        insertNewDataset(dataSource, dataSource.id.directoryName, dataStore).toFox
+          .map(Some(_)) // TODO: Discuss how to better handle this case
```

Comment:
Hmm, I see. Yes, it’s kind of unexpected. Unfortunately I don’t really have a good idea here. Since this was not introduced in this PR, I’d leave it as is for the time being.